main/src/sphexa/propagator.hpp

```diff
@@ -244,14 +247,38 @@ class HydroVeProp final : public Propagator<DomainType, ParticleDataType>
     MHolder_t mHolder_;
 
 public:
-    HydroVeProp(size_t ngmax, size_t ng0, std::ostream& output, size_t rank, bool doGravity)
-        : Base(ngmax, ng0, output, rank, doGravity)
+    HydroVeProp(size_t ngmax, size_t ng0, std::ostream& output, size_t rank)
+        : Base(ngmax, ng0, output, rank)
     {
     }
 
+    void activateFields(ParticleDataType& d) override
+    {
+        d.setConserved("x", "y", "z", "h", "m", "u", "vx", "vy", "vz", "x_m1", "y_m1", "z_m1", "du_m1", "alpha");
+        d.setDependent("p",
+                       "c",
+                       "ax",
+                       "ay",
+                       "az",
+                       "du",
+                       "c11",
+                       "c12",
+                       "c13",
+                       "c22",
+                       "c23",
+                       "c33",
+                       "xm",
+                       "kx",
+                       "divv",
+                       "curlv",
+                       "gradh",
+                       "keys",
+                       "nc");
+    }
+
     void sync(DomainType& domain, ParticleDataType& d) override
     {
-        if (doGravity_)
+        if (d.g != 0.0)
```

Comment:
Why do we still use a Universal Constant as a parameter, instead of a simple boolean that can be defined in the init, an input argument, or an input parameter file?